com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/dao/OperationDtoDao.java

```diff
@@ -104,6 +105,22 @@ public void merge(OperationDto operation) {
         });
     }
 
+    public OperationDto findFirstByMtaIdAndEndedAtIsNotNullOrderByEndedAtDesc(String mtaId) {
```

Comment:
The `OperationFilter` class exists exactly for this purpose - to avoid creating so specific methods, which are only used in one place. Use it.